extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -2304,4 +2272,71 @@ public static function onGetTalkPage( Title $title, Title &$talkPageTitle ) {
 		return true;
 	}
 
+	/**
+	 * SUS-845: Lock down Thread namespaces and Message Wall Greetings
+	 * For Thread namespaces, pnly allow edits to users with proper permission or to people editing their own posts
+	 * Everything else is disallowed.
+	 * For Wall greetings, only allow edits by owner or users with proper permission
+	 * @param Title $title
+	 * @param User $user
+	 * @param string $action
+	 * @param bool|string|string[] $result true or string or array of strings (MediaWiki error message key(s) to be shown if error)
+	 * @return bool Whether the user is allowed to perform this action or true if this is not a Wall/Forum title
+	 */
+	public static function onGetUserPermissionsErrors( Title $title, User $user, string $action, &$result ): bool {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this function "onGetUserPermissionsErrors" is 21 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)